File: Me.EarzuChan.Ryo/Utils/CompressionUtils.cs
using ICSharpCode.SharpZipLib.Zip.Compression;

namespace Me.EarzuChan.Ryo.Utils
{
    public static class CompressionUtils
    {
        // 是否抛出要看策略

        public static byte[] Inflate(byte[] input, int offset, int length)
        {
            try
            {
                var inflater = new Inflater();
                inflater.SetInput(input, offset, length);

                var buffer = new byte[1024];

                List<byte> outputList = [];

                while (!inflater.IsFinished)
                {
                    var count = inflater.Inflate(buffer);
                    outputList.AddRange(buffer.Take(count));
                }

                return outputList.ToArray();
            }
            catch (Exception e)
            {
                LogUtils.PrintError("解压出错", e);
                return [];
            }
        }

        public static byte[] Deflate(byte[] indexBytes, int offset, int length)
        {
            try
            {
                Deflater deflater = new();
                deflater.SetInput(indexBytes, offset, length);
                deflater.Finish();

                var buffer = new byte[1024];

                List<byte> outputList = [];

                while (!deflater.IsFinished)
                {
                    var count = deflater.Deflate(buffer);
                    outputList.AddRange(buffer.Take(count));
                }

                return outputList.ToArray();
            }
            catch (Exception e)
            {
                LogUtils.PrintError("压缩出错", e);
                return [];
            }
        }
    }
}

File: foundation/src/jvmMain/kotlin/me/earzuchan/ryo/foundation/util/CompressionUtils.kt
package me.earzuchan.ryo.foundation.util

import java.io.ByteArrayOutputStream
import java.util.zip.DataFormatException
import java.util.zip.Deflater
import java.util.zip.Inflater

actual object CompressionUtils {
    actual fun deflate(data: ByteArray): ByteArray {
        if (data.isEmpty()) return data

        val deflater = Deflater(Deflater.DEFAULT_COMPRESSION, false)
        deflater.setInput(data)
        deflater.finish()

        val out = ByteArrayOutputStream(data.size)
        val buffer = ByteArray(8192)
        while (!deflater.finished()) {
            val count = deflater.deflate(buffer)
            out.write(buffer, 0, count)
        }
        deflater.end()

        return out.toByteArray()
    }

    actual fun inflate(data: ByteArray): ByteArray {
        if (data.isEmpty()) return data

        val inflater = Inflater(false)
        inflater.setInput(data)

        val out = ByteArrayOutputStream(data.size * 2)
        val buffer = ByteArray(8192)

        try {
            while (!inflater.finished()) {
                val count = inflater.inflate(buffer)
                if (count > 0) {
                    out.write(buffer, 0, count)
                    continue
                }

                if (inflater.needsDictionary()) throw IllegalArgumentException("Inflater requires dictionary")
                if (inflater.needsInput()) throw IllegalArgumentException("Truncated deflated stream")
            }
        } catch (e: DataFormatException) {
            throw IllegalArgumentException("Invalid deflated data", e)
        } finally {
            inflater.end()
        }

        return out.toByteArray()
    }
}